lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1241,6 +1269,54 @@ def _parse_log_config(self):
         except ValueError as exc:
             self.fail('Error parsing logging options - %s' % (exc))
 
+    def _parse_healthcheck_options(self):
+        '''
+        Return healthcheck options
+        '''
+        if self.healthcheck.get('test') is None:
+            return None
+
+        if self.healthcheck.get('interval') is None:
+            self.healthcheck['interval'] = 30000000000  # (default: 30s)
+        else:
+            self.healthcheck['interval'] = self.convert_duration_to_nanosecond(self.healthcheck['interval'])
+
+        if self.healthcheck.get('timeout') is None:
+            self.healthcheck['timeout'] = 30000000000  # (default: 30s)
+        else:
+            self.healthcheck['timeout'] = self.convert_duration_to_nanosecond(self.healthcheck['timeout'])
+
+        if self.healthcheck.get('start_period') is None:
+            self.healthcheck['start_period'] = 0  # (default: 0s)
+        else:
+            self.healthcheck['start_period'] = self.convert_duration_to_nanosecond(self.healthcheck['start_period'])
+
+        if self.healthcheck.get('retries') is None:
+            self.healthcheck['retried'] = 3  # (default: 3)
+
+        return self.healthcheck
+
+    def convert_duration_to_nanosecond(self, time_str):
+        '''
+        Return time duration in nanosecond
+        '''
+        regex = re.compile(r'((?P<hours>\d+?)hr)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')
```

Comment:
Why don't you allow fractions of a second? (Not that it makes too much sense, but after all, docker expects nanoseconds.)